### PR TITLE
Changes 1

### PR DESCRIPTION
**Pull request by radhika011**
Changes made :-

In Principal Controller.java -
1) Use RESTful naming conventions for URLs  
2) Use @PathVariable annotation instead of @RequestParam annotation 
3) Use @Autowired to use the Bean 
4) Removing unnecessary imports

In AppConfig.java -
1) Add AppConfig.java file to define a Bean

In DemoApplication.java
1) Removing @RestController annotation
2) Removing unnecessary imports

### CODE REVIEW

File: AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.demo;
+import org.springframework.context.annotation.*;
+@Configuration
+public class AppConfig {
+	@Bean(initMethod = "init")
```
Comment (swapnildey94) [inline AppConfig.java]: @Bean is helpful when you have don't have access to class or you want to create lot of custom beans of the same class using different parameters. If you are using the default option - simply annotating the class with @Service / @Component will allow spring to automatically do this for you.

File: PrincipalController.java [hunk 2]
```diff
@@ -1,56 +1,51 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.*;
 import java.io.*;
 import org.springframework.stereotype.Component;
 @RestController
 
 public class PrincipalController {
-	ParkingLot pl1 = new ParkingLot();
+	@Autowired
+	ParkingLot pl1;
```
Comment (swapnildey94) [inline PrincipalController.java]: let's name our variable a little better - it makes the code more readable - change `pl1` to `parkingLot`

File: PrincipalController.java [hunk 3]
```diff
@@ -1,56 +1,51 @@
...
 	@RequestMapping()
 	String sayHello() {
 		return "Hello World!";
 	}
-	@RequestMapping(value="/park/id")
-	String park(@RequestParam("id") int carID,HttpServletResponse response) throws IOException{
-		Car car1 = new Car(carID);
+	@RequestMapping(value="/parking-lot/car/car-id/{car_id}/park")
+	String park(@PathVariable int car_id,HttpServletResponse response) throws IOException{
+		Car car1 = new Car(car_id);
```
Comment (swapnildey94) [inline PrincipalController.java]: same here `car1` to `car`

File: PrincipalController.java [hunk 4]
```diff
@@ -1,56 +1,51 @@
...
 		int slotNo = car1.park(pl1);
 		if(slotNo==-1) {
 			response.sendError(HttpStatus.BAD_REQUEST.value());
 		}
-		return  "Car "+carID+" parked at "+slotNo;
+		return  "Car "+car_id+" parked at "+slotNo;
```
Comment (swapnildey94) [inline PrincipalController.java]: read about StringBuilder class that Java 8 provides and how you can use it here

File: PrincipalController.java [hunk 5]
```diff
@@ -1,56 +1,51 @@
...
 	}
-	@RequestMapping(value="/unpark/slot")
-	String unpark(@RequestParam("slot") int slotNo,HttpServletResponse response) throws IOException{
-		Car unparked = pl1.unPark(slotNo);
+	@RequestMapping(value="/parking-lot/slot/slot-no/{slot_no}/unpark")
```
Comment (swapnildey94) [inline PrincipalController.java]: I prefer camel case for url definitions as we can avoid usage of special characters `-` -> `parking-lot` to `parkingLot`

File: PrincipalController.java [hunk 6]
```diff
@@ -1,56 +1,51 @@
...
+	String unpark(@PathVariable int slot_no,HttpServletResponse response) throws IOException{
+		Car unparked = pl1.unPark(slot_no);
 		if(unparked==null) {
 			response.sendError(HttpStatus.BAD_REQUEST.value());
 		}
-		return "Car with car ID "+unparked.getCarID()+" unparked from "+slotNo;
+		return "Car with car ID "+unparked.getCarID()+" unparked from "+slot_no;
```
Comment (swapnildey94) [inline PrincipalController.java]: string builder

**Review by swapnildey94 — CHANGES_REQUESTED** [review]
Given a few more pointers and feedback about how we can improve the code here. I want you to now check out spring boot data package and see how we can integrate MongoDB with this?

**Comment by radhika011** [comment]
In PrincipalController.java -
1) Changing variable names - pl1 to parkingLot and car1 to car
2) Using StringBuilder to optimize string append operations
3) Using camelCase in URLs and other variables

Deleting AppConfig.java

In ParkingLot.java
1) Adding @Service annotation

**Review by swapnildey94 — APPROVED** [review]

**Pull request merged**